### ISSUE TITLE
Add the /debug:portable switch and MSBuild support

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#5534]
Right no the debug switch supports:

- none
- pdbonly 
- full

There needs to be the additional value of portable for the new portable PDB format. 

Split out from #5285.  The original bug will cover the VS support while this bug will cover the compiler and MSBuild.

**Comment by TyOverby** [comment]
It seems like right now we just have two options 

```
emitPdb is 
  "none" -> false,
  "pdbonly" -> true,
  "full" -> true,
```

To be clear, I'm supposed to add a "fourth" option `portable` and a "third" value which will tell the compiler to emit portable PDB files?

**Comment by tmat** [comment]
The following table:

---
option | bool emitPdb | DebugInformationFormat debugInfoFormat |
--- | --- | --- |
/debug- | false | Pdb |
/debug+ | true | Pdb |
/debug:pdbonly | true | Pdb |
/debug:full | true | Pdb |
/debug:portabe | true | PortablePdb |
/debug:embedded | true | Embedded |

Although I would consider not implementing /debug:embedded for U1 since it's not well tested.

**Comment by tmat** [comment]
@jaredpar Does that sound right?

**Comment by TyOverby** [comment]
@tmat Where does this DebugInformationFormat end up going? Once I parse it out of the command line, where do I send it so that the backend knows how do spit out the right PDB format?

**Comment by tmat** [comment]
It's set to EmitOptions. You can find the code that extracts it from /features in the current CSharpCommandLineParser.

**Comment by tmat** [comment]
(Once the /debug switch is in, you can remove said code)

**Comment by TyOverby** [comment]
To be clear, I'm also adding this to the VB compiler right?

**Comment by jaredpar** [comment]
This is expanding beyond the original scope.  The intent here is to add /debug:portable only.

**Comment by tmat** [comment]
@TyOverby Yes, VB should have the same option.

**Comment by TyOverby** [comment]
> This is expanding beyond the original scope

In what way?

**Comment by TyOverby** [comment]
Closed by #5600
